### DISCUSSION TITLE
Please remain neutral

### DISCUSSION THREAD

**Discussion by ToulthG** [starship/starship#3755] [General]
:)

**Comment by matchai** [comment]
The FOSS movement is a political act. If you'd like to stay out of politics, use for-profit software.

Maintainers and contributors of open source software that we depend on every day are having their lives uprooted and are being killed. If you don't like facing that reality, just scroll past the banner.

**Reply by igorsaux** [reply]
> The FOSS movement is a political act.

And how soon we can expect of pushing a malware in this repo as a political act?

P.S. Please, don't speak for FOSS. Your actions are not welcome there and FOSS is not about policital propaganda like you do.

**Comment by philippetev** [comment]
There is a good reason most specialized forums to have forbidden the politics in any form in their rules. I see that now...

**Reply by matchai** [reply]
Forbidding the discussion of politics directly benefits those in a privileged position by allowing them to maintain the status quo. It exclusively harms the underprivileged and no one else.

**Reply by philippetev** [reply]
> Forbidding the discussion of politics directly benefits those in a privileged position by allowing them to maintain the status quo. It exclusively harms the underprivileged and no one else.

They're forbidding politics simply because it has NOTHING to do with their forums, not because someone has told them to maintain his status quo.
What kind of a business has politics in computer/tech or, let's say, car-specific forums? Exactly?
Because that's the kind of "specialized forums" I was talking about, in case you have missed that part. Such as this place...

In any specialized political or general-purpose forums - sure, why not. Very often in the "offtopic" subcategories...

And there are also 4 more ongoing major armed conflicts worldwide, along with several minor, in case you choose to be selective...

**Reply by matchai** [reply]
Yes, we're being selective. Maintainers in our ecosystem are being very directly impacted by the ongoing war in Ukraine. Our friends building [Spaceship ZSH](https://github.com/spaceship-prompt/spaceship-prompt) are experiencing this firsthand.

This isn't the first time we've had a political stance and this certainly won't be the last.

You're welcome to use alternatives if you're dissatisfied with our political stance:
- https://github.com/JanDeDobbeleer/oh-my-posh
- https://github.com/liamg/bearings
- https://github.com/romkatv/powerlevel10k
- https://github.com/IlanCosman/tide

**Reply by philippetev** [reply]
> You're welcome to use alternatives if you're dissatisfied with our political stance:
> 
>     * https://github.com/JanDeDobbeleer/oh-my-posh
> 
>     * https://github.com/liamg/bearings
> 
>     * https://github.com/romkatv/powerlevel10k
> 
>     * https://github.com/IlanCosman/tide

Why would I do that, if I'm not displeased with the product itself? Simply because of some minor disagreements on another topic with one of its coders?
I think you're confusing me with someone else (yourself maybe?)...

Please, don't bother to answer me. This conversation is over, for me at least.

**Reply by bkerin** [reply]
You're mooching spectacularly off all contributors who *don't* endorse the US project in Ukraine (many because they feel the ongoing militancy there is getting a lot of ordinary Ukrainians killed for no particularly good reason).

**Reply by webknjaz** [reply]
And... Unsubstantiated nonsense strikes again 🤦🏼

May you'd consider to stop spreading terrorist propaganda instead?

**Reply by bkerin** [reply]
Are you claiming you *did* chekc with all 400+ contributors and they all approved the banner?  Because if not you're the one pumping on garbage here.

It would be the same if it was a big russian flag with GO PUTIN blazoned all over it btw.

"stop spreading terrorist propaganda" ffs just stop being a total jackass

**Reply by webknjaz** [reply]
It's up to the maintainers' to be decent just like it is up to you not to be. I only see you spreading the talking points of mass murderers. If you act like a ruscist, expect duck typing to be applied. So it's you who should stop supporting the terrorists, okay?

**Comment by TheTechRobo** [comment]
"Please remain neutral" the developers _would_, if Russia remained neutral and stopped the invasions.

It goes both ways.

**Comment by oblitum** [comment]
> The FOSS movement is a political act. If you'd like to stay out of politics, use for-profit software.

This is _very_ rude, just as it's incorrect:

* FOSS **can** be for-profit (it's a common misconception it cannot)
* [For-profit can be political](https://www.linux.com/news/microsofts-ballmer-linux-cancer/)
* What political act/movement FOSS was funded into? Copyleft ideals! `starship` is copyright! (so that argumentation line isn't even worth carrying on...)
* Software ownership politics is a complete different beast than geopolitics, to remain or not remain neutral on the latter has absolutely no relation with the stance of the former
  * If you wish to bring other kinds of politics into this software you might change your license, for example, there's the Anti-996 license that endorses the Anti-996 movement, which entails work politics. Maybe there's an Anti-Russia or Anti-Putin license...
* (Consciously) not taking (given) sides can also be a political act

> Maintainers and contributors of open source software that we depend on every day are having their lives uprooted and are being killed. If you don't like facing that reality, just scroll past the banner.

Donbass people have been shelled for damn long eight years (and you kept silent), Russia rightfully (but too late) recognized their republics' _independency_, before they or their culture were completely decimated (notice though that endorsing independence doesn't equate to endorsing war).

If I (as user/contributor) lived there I'd be upset about your stance, but at the same time I'd probably do nothing about, given the feeling of impotence that fighting western narratives can bring.

Have you gone beyond western main stream media talking points analysis? Checked Donbass situation since [the US-funded colored revolution Euromaidan took place](https://www.quora.com/Why-in-March-2022-did-YouTube-remove-Oliver-Stones-2016-documentary-Ukraine-on-Fire) (2014)? What about the Minsk agreements? Or the extent of [neo-nazi infiltration](https://www.youtube.com/watch?v=KfaAyiP8Wuc) in Ukraine? Etc...

Given all _that_, I'd like to state _that_ is not about that I'm taking a side, Russia, Ukraine, Donbass' republics, it's about not attaching one where it's not due, the software, and by consequence, to all its users **and** contributors, instead of a _personal_ blog post of yours, which happens to be the original author.

What's happening there, to real people, really depresses me (just like in any of the other several conflicts generated by US meddling, [including where I live](https://library.brown.edu/create/wecannotremainsilent/chapters/chapter-1-revolution-and-counterrevolution-in-brazil/the-u-s-government-and-the-1964-coup/), including [in the US itself](https://www.youtube.com/watch?v=Vy6lFFaX0nQ)).

Now, changing topics a bit and entering conspiracies... I've seen this happening in Rust already, which is spreading to Rust-based projects, it's stronger than elsewhere in general. I've seen the same fallacious argument of conflating politics of code ownership to geopolitics, to code of conduct, etc., coming from Rust developers. This (conflation) is wrong. I can notice it's in their culture for a long time, say, starting with strong codes of conduct a long time ago for shutting down discussions, etc., and which is evolving to geopolitical stances. I wonder whether maybe @steveklabnik had a point when it "freaked out" on that Amazon/Rust drama, I thought it was just a drama but... this is actually fertile ground for a big US corp to take advantage of.

Anyway, this is [the nail in the coffin for starship to me](https://codeberg.org/oblitum/dotfiles/commit/e488c8c40465766bb7398e24fc86d8330ba3d3c9), I ended up landing here while innocently looking for the current state of async... and then have hit "that banner" on landing page. My sole lasting request, as a once very minor contributor, is just that you respect this comment and don't delete/edit it.

EDIT:

I've since checked the banner's link content and consider that page of very bad taste due to the pervasive cancel culture of anything Russian that it promotes. This solely creates stronger divisions and hate promotion, besides being childish.

**Reply by webknjaz** [reply]
> Donbass people have been shelled for damn long eight years

You forgot that they've been shelled by r*ssia. And stating otherwise is a typical tactic of their nazi propaganda. Are you a neonazi too? Your statements align with the ruscist ones. So I suppose the answer is "yes" since you've repeated almost _every_ propagandist pro-war excuse to commit genocide on the Ukrainian land (including Donbas). Anyway, it's a rhetoric question. I see what you are, others probably too.

**Reply by oblitum** [reply]
You're lying and trolling the thread. Not helpful.

**Reply by philippetev** [reply]
> You forgot that they've been shelled by r*ssia. 

Please, provide any proves for such claims/accusations. Or get lost. It's as simple as that.
Two options for you, but considering the fact we are at code repository, not involved with politics in any way and therefore, such topics have no place, the second one will be more proper.

**Reply by webknjaz** [reply]
> You're lying and trolling the thread

Not at all. But you are for some reason. In accordance to duck typing, you are a ruscist.

**Reply by webknjaz** [reply]
> Please, provide any proves for such claims/accusations. Or get lost

I don't owe you anything, especially not with this attitude. But I have proof enough. My family was in an occupied village close to Kyiv, they've faced the ruscists (as in had a tank pointing at their house for a week and had belongings taken from them). Luckily they've survived, many didn't.
What's your excuse for supporting r*ssian nazis?

**Reply by philippetev** [reply]
@webknjaz Exactly what part of "Get lost" you didn't get?

**Comment by microyahoo** [comment]
@matchai Please don't impose your own ideology on other people. The code is neutral, why should you add something unrelated to the code? Have you considered the feelings of the people of North Korea, Vietnam, Iraq, Afghanistan, Syria, and Libya? If you don't understand, please SHUT UP.

**Reply by webknjaz** [reply]
> please SHUT UP.

You too. r*ssian nazi :ship: you know where to go.

https://en.wikipedia.org/wiki/Whataboutism

**Reply by microyahoo** [reply]
Wake up, look at what you've done, what freedom and human rights, look at what you've done to those people (North Korea, Vietnam, Iraq, Afghanistan, Syria, and Libya).

**Reply by microyahoo** [reply]
I'm neutral. You don't reflect on what you've done, but instead fan the flames everywhere

**Reply by microyahoo** [reply]
You can express your opinion, but please do not reflect it in the code

**Reply by philippetev** [reply]
Hey, @webknjaz, in case you (obviously) have forgotten, GitHub is a place for code and sharing code. Pseudo-"projects" like yours (_**not just about Ukraine, but about ANY other conflict involving politics/geopolitics**_) have no place here. I wonder why GitHub haven't forbidden politics in any form in their policy yet...

So, you and people like you, go find any general-purpose or even political forum and spread your political pollution (BS) there!

And next time when you call someone "russian", at least check first where is he/she really from (quick tip: it's usually written).
If you don't know where Bulgaria on the map is, speaking something else is usually considered as a bug in your TV, or [Words ahead of thought](https://encyclopedia2.thefreedictionary.com/First-Signal+System) as we say in Bulgaria.

**Reply by webknjaz** [reply]
> I'm neutral

Yet, your behavior shows otherwise. You just posted a number of ruscist propaganda in support of the genocide (yet another one).

**Reply by webknjaz** [reply]
> I wonder why GitHub haven't forbidden politics in any form in their policy yet...

Is this why you're making posts in support of nazis and genocide?

**Reply by webknjaz** [reply]
> And next time when you call someone "russian", at least check first where is he/she really from

Oh, I did. And I correctly identified a ruscist in front of me. Nazi supporters don't exclusively live on some territory, I saw a ton of emigrants spreading the r*ssian propaganda, just like you.
Also, I closed the issue because it was doing the same as many other propagandist bots/trolls in other issues. In reality, it doesn't really matter where a person is from if their behavior identifies with a certain known toxic behavior.
This is why it was closed. You behaved like a ruscist, you were supporting r*ssian nazis, spreading their propaganda. Do this at your home, not in civilized places.

**Reply by philippetev** [reply]
You did $hit.
So now talking against ANY political involvement of ANY kind in GitHub is classified as "toxic behavior"? Seriously ?!? Yeah, sure. Whatever...

**Comment by sanblch** [comment]
@matchai, I advice an [alternative](https://github.com/AndrewStetsenko/Support-Ukraine) for the banner you've placed. I think it's more general, it gathers many sources and makes more objective prospective on the war. Moreover it doesn't suggest funding the war.

**Reply by oblitum** [reply]
Not great... that alternative is endorsing a [no-fly zone by NATO...](https://www.youtube.com/watch?v=YSxnyA_8RI0)

**Reply by webknjaz** [reply]
> more objective prospective on the war

More objective then the one by the people being shelled? :thinking: Or boi. You ruscists truly got no shame.

**Reply by sanblch** [reply]
@webknjaz, Please, speak like a man. I don't think you are shelled. You are just making propaganda. Your banner states as fact, but doesn't provide with information.

**Reply by sanblch** [reply]
Yeah, there are some links. Hope you chose them right. Because ukraine army crimes are plenty.

**Reply by sanblch** [reply]
Video on the links is fake. Photos are random, but probably military dislocations in the cities. Typical agenda.

**Reply by webknjaz** [reply]
> Please, speak like a man.

I do. You do not.

> I don't think you are shelled.

Not personally/physically. But my close family is.

> You are just making propaganda. Your banner states as fact, but doesn't provide with information.

The fact that you can even bother to use google to see that you are lying to yourself is louder than words. We are not making up anything, all is coming from the trusted sources which is not something I could say about you.

**Reply by webknjaz** [reply]
> Video on the links is fake.

Said a nazi apologist... What a coincidence. Repeating r*ssian propaganda does not make you believable at all.

**Reply by EpsilonKu** [reply]
@webknjaz Pls, stop saying to everyone that they are nazi. Saying someone nazi because suggesting more general banner. War is shit, there will be casualties no matter what. Syria 3847 civilians by Coalition force.  Yeah, maybe you will also say that "You are also Nazi" and not analysing anything.Btw, I'm not Russian.

**Reply by webknjaz** [reply]
This has nothing to do with the national origin or location. Only with the behavior of an individual. If you're familiar with duck typing, that's exactly it. If one mimics the behavior of the nazi propaganda, they would be identified as belonging to the same category, that's it, nothing more.

P.S. Whataboutism is a widely used technique ruscists (in particular) use. This may identify you as a genocide supporter following the same principle.

**Reply by EpsilonKu** [reply]
I don't think that generalization all people as Nazi(not dependent on nation) is correct. You are not respecting any opposite opinion. No one supporting Putin regime.

**Reply by webknjaz** [reply]
> I don't think that generalization all people as Nazi(not dependent on nation) is correct. You are not respecting any opposite opinion. No one supporting Putin regime. 

It's statements like this that the regime relies on to exist, to operate, to be possible. If not active, it's definitely a passive support.
Besides, it's a propagandist tale that only a single person is responsible. Just think about it — is it putin personally murdering people, raping, killing kids? No, the disciples do this. Where did they learn to behave like this? Well, in the society. Are they all coming from one specific location/town? Nope, they represent an average r*ssian. It is quite evident what the society is made of by just looking at the war criminals, listening to their home call intercepts. Alright, who made a whole society like this? The previous generation. Not one generation, quite a few. R*ssians have been taking steps to undermine, forbid and wipe the Ukrainian culture, language, self-identity for several hundreds of years. They issued decrees specifically forbidding the use of the language, forcefully relocated tons of people, committed genocided (3 Holodomors in just the 20th century, off the top of my head, millions killed by starvation).
It looks like you didn't take any steps to understand what's actually happening. The context matters. The ruscists are bad. But it's not a a single generation that's like this. Multiple generations for hundreds of years grew up in the ukrainophobic environment. It's a cultural thing. They also have a liberator's complex. So yes, there's around 0.001% of people who are not comfortable with what is being committed in their name. The rest are fine living as they live, being genocide enablers. They usually don't like to think about what's happening to someone else and they live a happy life pretending that the responsibility is somebody else's. But the responsibility is shared.
Until the society changes and a few generations passes, there's no way the perception of this terrorist state changes.

**Comment by igortas** [comment]
Macedonia and Ukraine are two Christians countries, that suffer the most, through their history and they have similarities in their struggles. While today everyone is losing their mind with Ukraine, like they really care (they don't) no one even mentioned that Macedonia recently, lost their sovereignty country name 'Republic of Macedonia', by interfering into their politics, directly from USA , UK, nearly whole EU union, and few satellite countries, and with just few checks on google that looks like legitimate, approved from the people living there (also Macedonians haved similar kind of small wars in  2001 and 2014), and now the country is called North Macedonia (which people there rejects and they live in apathy, yeah developers too), that also means people lost their nationality, which is damn worse than the war in Ukraine. It's just matter of time when that country will vanish (or the whole country structure will change), will be divided, and no knows will ever be remember it, in the near future. So please developers, when next time put your crap banner on your site, please also support other countries.

**Reply by webknjaz** [reply]
And... Whataboutism strikes again. Please, stop this ruscism fight.

**Reply by igortas** [reply]
> And... Whataboutism strikes again. Please, stop this ruscism fight.
u, bunch of liberal developers can't do anything about ukraine, and that's a shame.

**Reply by bkerin** [reply]
"whataboutism" is a bullshit term that people use to avoid the cognitive dissonance they get when the utter hypocrisy of their positions is made indisputably clear by an example

**Reply by webknjaz** [reply]
Oh, boi. Not even common sense can help you ruscists.

**Reply by MaxGraey** [reply]
Comparing the conflict with Greece, which caused the country to be renamed, come to a consensus and join NATO in 2020 with the biggest military conflict in Europe since WWII between the two biggest countries on the content. You're so fucking selfish

**Reply by matchai** [reply]
Personal harassment and ableist language is not welcome here and directly breaches our code of conduct. A ban was placed on this user's account from interacting in the Starship community.

**Edit:** Looks like the account has since been deleted.

**Reply by igortas** [reply]
> Comparing the conflict with Greece, which caused the country to be renamed, come to a consensus and join NATO in 2020 with the biggest military conflict in Europe since WWII between the two biggest countries on the content. You're so fucking selfish

No consensus retard at all, 35% or less.... First try to find relevant info that talk... please stop talk for yourself... liberals government dictated by murica wishesh... go fund raise ukraine ahahaha...

**Reply by igortas** [reply]
> Personal harassment and ableist language is not welcome here and directly breaches our code of conduct. A ban was placed on this user's account from interacting in the Starship community.
> 
> **Edit:** Looks like the account has since been deleted.

when lies can't become truth, ban those who oppose.

**Reply by matchai** [reply]
As was just said, ableist language is not tolerated here and directly breaches our code of conduct.  We haven't been interfering with this discussion, but your use of ableist slurs right after a user was warned shows that you're not here arguing in good faith.

You've been similarly banned from interacting with the Starship community.

**Reply by bkerin** [reply]
You're *not* a moron, obviously, and I would never not say you are.  So you understand the situation here.  So how the hell do you convince yourself that you're not a huge mooch for claiming to speak for your entire dev community with those banners?  I'm honestly curious

**Comment by argenkiwi** [comment]
Now that the [banner has been removed ](https://github.com/starship/starship/commit/099539fd1aa2937e9507a318becffcbc685b83b8), will the project remain neutral or should we expect one about the conflict between Israel and Palestine to be added soon?

**Reply by webknjaz** [reply]
It's never neutral, it just gave in to the ruscist terrorist platform, that's about it.
